lib/ide/src/transport/test_utils.rs

```diff
@@ -0,0 +1,45 @@
+use crate::prelude::*;
+
+use crate::executor::test_utils::TestWithLocalPoolExecutor;
+
+use json_rpc::test_util::transport::mock::MockTransport;
+
+
+
+#[derive(Debug)]
+pub struct TestWithMockedTransport {
+    with_executor_fixture : TestWithLocalPoolExecutor,
+    transport             : MockTransport,
+    next_response_id      : usize,
+}
+
+impl TestWithMockedTransport {
+    pub fn set_up(transport:&MockTransport) -> Self {
+        Self {
+            with_executor_fixture : TestWithLocalPoolExecutor::set_up(),
+            transport             : transport.clone_ref(),
+            next_response_id      : 0,
+        }
+    }
+
+    pub fn run_test<TestBody>(&mut self, test:TestBody) -> &mut Self
+    where TestBody : Future<Output=()> + 'static {
+        self.with_executor_fixture.run_test(test);
+        self
```

Comment:
Is it really useful to return self in all these methods?
Chaining is mostly useful when dealing with temporaries (like ad-hoc function arguments), but I don't think this is really the case with our test runners.